src/client/java/hadoop-daos/src/main/java/io/daos/fs/hadoop/DaosInputStream.java

```diff
@@ -103,12 +103,8 @@ public DaosInputStream(DaosFile daosFile,
     this.buffer.limit(0);
     this.fileLen = daosFile.length();
     this.bufferCapacity = buffer.capacity();
-    this.preLoadSize = preLoadSize;
+    this.preLoadSize = bufferCapacity < preLoadSize ? bufferCapacity : preLoadSize;
```

Comment:
The assumption that preLoadSize needs to be less than bufferCapacity is confusing. Can we remove preLoadSize and only use a single buffer size in DaosInputStream constructor. It always uses the buffer size to read the data from DAOS. If  the application create DaosInputStream with a small buffer size, we can just use the predefined default read buffer size instead.

Reply:
@carsonwang please review the change code.

Reply:
As tested against workload repartition with scale of bigdata, the perf is same as before. 
Without preload, I got below result.
|min-read size|perf|
|---------------|----|
|64KB|5,120 MB|
|128KB|5,167MB|
|512KB|5,214MB|
|1024KB|5261MB|

With preload
|preload size|perf|
|-------------|----|
|1024KB|5167MB|